### PR TITLE
Add scale param to images

### PR DESCRIPTION
**Pull request by chriddyp**
check it @cldougl @aneda @theengineear @BRONSOLO

### CODE REVIEW

File: plotly/tests/test_core/test_image/test_image.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            # Shouldn't need to skip these tests, the server should
+            # be responding with a 400 level error since scale isn't supported,
+            # but it doesn't yet, so just skip them
+            continue
+
+        _test = test_generator(*args)
+        arg_string = ', '.join([str(a) for a in args])
+        test_name = test_generator.__name__.replace('_generate', 'test')
+        test_name += '({})'.format(arg_string)
+        setattr(TestImage, test_name, _test)
```
Comment (chriddyp) [inline plotly/tests/test_core/test_image/test_image.py]: this is so much fun. 

autogenerating tests of all of the permutations of input arguments!

```
♡ nosetests plotly/tests/test_core/test_image/ --pdb -v
_test_image_get_returns_valid_image_test(jpeg, 300, 300, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(jpeg, 300, None, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(jpeg, None, 300, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(jpeg, None, None, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(pdf, 300, 300, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(pdf, 300, None, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(pdf, None, 300, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(pdf, None, None, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(png, 300, 300, 5) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(png, 300, 300, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(png, 300, None, 5) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(png, 300, None, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(png, None, 300, 5) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(png, None, 300, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(png, None, None, 5) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(png, None, None, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(svg, 300, 300, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(svg, 300, None, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(svg, None, 300, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
_test_image_get_returns_valid_image_test(svg, None, None, None) (plotly.tests.test_core.test_image.test_image.TestImage) ... ok
```

File: plotly/tests/test_core/test_image/test_image.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        self.data = [{'x': [1, 2, 3], 'y': [3, 1, 6]}]
+
+
+def _generate_image_get_returns_valid_image_test(image_format,
+                                                 width, height, scale):
+    def test(self):
+        image = py.image.get(self.data, image_format, width, height, scale)
+        if image_format in ['png', 'jpeg']:
+            assert imghdr.what('', image) == image_format
```
Comment (chriddyp) [inline plotly/tests/test_core/test_image/test_image.py]: have i ever told you how much i like the python standard library?

File: plotly/tests/test_core/test_image/test_image.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+kwargs = {
+    'format': ['png', 'jpeg', 'pdf', 'svg'],
+    'width': [None, 300],
+    'height': [None, 300],
+    'scale': [None, 5]
+}
+
+for args in itertools.product(kwargs['format'], kwargs['width'],
+                              kwargs['height'], kwargs['scale']):
```
Comment (chriddyp) [inline plotly/tests/test_core/test_image/test_image.py]: all of the possible combinations of the kwargs, e.g. 

```
('png', None, None, None)
('png', None, None, 5)
('png', None, 300, None)
('png', None, 300, 5)
('png', 300, None, None)
('png', 300, None, 5)
('png', 300, 300, None)
('png', 300, 300, 5)
('jpeg', None, None, None)
('jpeg', None, None, 5)
('jpeg', None, 300, None)
('jpeg', None, 300, 5)
('jpeg', 300, None, None)
('jpeg', 300, None, 5)
('jpeg', 300, 300, None)
('jpeg', 300, 300, 5)
('pdf', None, None, None)
('pdf', None, None, 5)
('pdf', None, 300, None)
('pdf', None, 300, 5)
('pdf', 300, None, None)
('pdf', 300, None, 5)
('pdf', 300, 300, None)
('pdf', 300, 300, 5)
('svg', None, None, None)
('svg', None, None, 5)
('svg', None, 300, None)
('svg', None, 300, 5)
('svg', 300, None, None)
('svg', 300, None, 5)
('svg', 300, 300, None)
('svg', 300, 300, 5)
```

File: plotly/tests/test_core/test_image/test_image.py (hunk 2 above)
Comment (theengineear) [inline plotly/tests/test_core/test_image/test_image.py]: haha, yeah that's a great one!

**Comment by theengineear** [comment]
@chriddyp , yup, it'd be best to test that it's 4xx for now.

**Comment by theengineear** [comment]
this is sweet btw! :dancer:

**Comment by BRONSOLO** [comment]
:+1:

**Comment by chriddyp** [comment]
@theengineear @BRONSOLO - it's not supposed to be a 4xx though, it's supposed to be a 200!

**Comment by theengineear** [comment]
@chriddyp sorry read that too fast, looking into it now (https://github.com/plotly/python-api/pull/266#issuecomment-125243770)

**Pull request merged**